src/Http/WebUtilities/src/MultipartPipeReader.cs

```diff
@@ -0,0 +1,345 @@
+using System;
+using System.Buffers;
+using System.Diagnostics;
+using System.IO;
+using System.IO.Pipelines;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Internal;
+
+namespace Microsoft.AspNetCore.WebUtilities
+{
+    public class MultipartPipeReader
+    {
+
+        public const int DefaultHeadersCountLimit = 16;
+        public const int DefaultHeadersLengthLimit = 1024 * 16;
+        //private const int DefaultBufferSize = 1024 * 4;
+        private const int StackAllocThreshold = 128;
+
+        private readonly PipeReader _pipeReader;
+        //private ReadOnlySequence<byte> _buffer;
+        private long _bytesConsumed = 0;
+        private readonly MultipartBoundary _boundary;
+        private MultipartPipeReaderStream _currentStream;
+
+        private static readonly byte[] ColonDelimiter = new byte[] { (byte)':' };
+        private static readonly byte[] CrlfDelimiter = new byte[] { (byte)'\r', (byte)'\n' };
```

Comment:
Write them as ROSpan --> https://github.com/aspnet/AspNetCore/pull/15056#discussion_r336611339
So the static data (in the assembly) is accessed directly, without any copying.

Reply:
By making it a static field I can't make use ROSpan, I get the following error

>Field or auto-implemented property cannot be of type 'ReadOnlySpan<byte>' unless it is an instance member of a ref struct.

I think this is an issue I wasn't familiar with, does sending a byte[] to TryReadTo have some performance issues? I thought that by sending a pointer to array there should have been no copies - or does it create a new byte on the stack for every byte in the array that you want me to avoid? 

I guess I can just send them to the function with .AsSpan()

Reply:
```suggestion
        private static ReadOnlySpan<byte> CrlfDelimiter => new byte[] { (byte)'\r', (byte)'\n' };
```
For background see [C# ReadOnlySpan and static data](https://vcsjones.dev/2019/02/01/csharp-readonly-span-bytes-static/).